src/main/scala/ai/privado/passes/DBTParserPass.scala

```diff
@@ -0,0 +1,462 @@
+    storeForTagHelper(Constants.id, ruleInfo.id)
+    storeForTagHelper(Constants.nodeType, ruleInfo.nodeType.toString)
+    storeForTagHelper(Constants.catLevelOne, ruleInfo.catLevelOne.name)
+    storeForTagHelper(Constants.catLevelTwo, ruleInfo.catLevelTwo)
+
+    databaseNode
+  }
+
+  private def storeForTag(
+    builder: BatchedUpdate.DiffGraphBuilder,
+    node: AstNodeNew
+  )(tagName: String, tagValue: String = ""): BatchedUpdate.DiffGraphBuilder = {
+    builder.addEdge(node, NewTag().name(tagName).value(tagValue), EdgeTypes.TAGGED_BY)
+    builder
+  }
+
+  private def createTableColumnNodesFromModels(
+    builder: DiffGraphBuilder,
+    model: java.util.Map[String, Any],
+    tableQueryOrder: Int
+  ) = {
+    val modelFile = model.get("filePath").asInstanceOf[String]
+    val tableName = model.get("name").asInstanceOf[String]
+
+    // create file node
+    val fileNode = addFileNode(modelFile, builder)
+
+    // create table node
+    val (tableLineNumber, tableColumnNumber, tableMatchedLine) =
+      findFirstPatternInYAMLFile(modelFile, "name", tableName)
+    val tableNode = NewSqlTableNode()
+      .name(tableName)
+      .code(tableMatchedLine)
+      .lineNumber(tableLineNumber)
+      .columnNumber(tableColumnNumber)
+      .order(tableQueryOrder)
+
+    builder.addNode(tableNode)
+    builder.addEdge(tableNode, fileNode, EdgeTypes.SOURCE_FILE)
+
+    // create column nodes
+    val columns = model.get("columns").asInstanceOf[java.util.List[java.util.Map[String, Any]]]
+    columns.asScala.zipWithIndex.foreach { case (column, index) =>
+      val colName                                          = column.get("name").asInstanceOf[String]
+      val (colLineNumber, colColumnNumber, colMatchedLine) = findFirstPatternInYAMLFile(modelFile, "name", colName)
+      val columnNode = NewSqlColumnNode()
+        .name(colName)
+        .code(colMatchedLine)
+        .lineNumber(colLineNumber)
+        .columnNumber(colColumnNumber)
+        .order(index)
+
+      logger.debug(f"Creating column node for $colName in table node $tableName")
+
+      // link nodes
+      builder.addNode(columnNode)
+      builder.addEdge(tableNode, columnNode, EdgeTypes.AST)
+      builder.addEdge(columnNode, fileNode, EdgeTypes.SOURCE_FILE)
+    }
+
+    tableNode
+  }
+
+  private def findFirstPatternInYAMLFile(filePath: String, key: String, value: String): (Int, Int, String) = {
+    val (escapedKey, escapedValue) = (Regex.quote(key), Regex.quote(value))
+    val pattern                    = s"""["']?$escapedKey["']?\\s*:\\s*["']?$escapedValue["']?\\s*$$""".r
+    val lines                      = Files.readAllLines(Paths.get(filePath)).toArray(Array.ofDim[String](0))
+
+    val (defaultRow, defaultCol) = (1, -1)
+
+    lines.zipWithIndex
+      .flatMap { case (line, rowIndex) =>
+        val matches = pattern.findFirstMatchIn(line)
+        matches.map(matchResult => (rowIndex + 1, matchResult.start + 1, matchResult.group(0)))
```

Comment:
In this case, since group(0) is inside `map` from `matches`, there will be a match result, where 0th index refers to the entire match.
In case there is no match, the subsequent `getOrElse` is triggered.